pkg/registry/generic/etcd/etcd.go

```diff
@@ -301,7 +301,7 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 		}
 		if newVersion != version {
 			// TODO: return the most recent version to a client?
-			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("the resource was updated to %d", version))
+			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("another client made an update to this object in resource version %d", version))
```

Comment:
----- Original Message -----
> > @@ -301,7 +301,7 @@ func (e *Etcd) Update(ctx api.Context, obj
> > runtime.Object) (runtime.Object, bool
> >  		}
> >  		if newVersion != version {
> >  			// TODO: return the most recent version to a client?
> > -			return nil, nil, kubeerr.NewConflict(e.EndpointName, name,
> > fmt.Errorf("the resource was updated to %d", version))
> > +			return nil, nil, kubeerr.NewConflict(e.EndpointName, name,
> > fmt.Errorf("another client made an update to this object in resource
> > version %d", version))
> 
> There's two problems with the wording:
> * apiserver doesn't know if it was another client that made the conflicting
> change (it probably is, but we shouldn't report it as certain-- imagine a
> user modifying a pod and then modifying it again without re-getting it: this
> error will be very confusing for them.)
> * "made an update to this object in resource version X" is grammatically
> confusing and hard to parse. And by my parsing, incorrect-- the first
> offending change would actually be resourceVersion X+1.
> 
> I agree we can do better than the old wording, and that making this clear is
> a good goal, but I don't agree that the new wording will actually help
> people more than it confuses them.
> 
> I actually think my last suggestion ("the resourceVersion you reported (%v)
> is out-of-date; please re-read the object and try your change again") is
> best. Maybe amended to "the resourceVersion you reported (%v) is
> out-of-date, the object has been modified; please re-read the object and try
> your change again"

Some of that is almost like instructions to a programmer, and also long.  Also, %v was the new resource version, not the old one, so the user could copy and paste it into whatever they're changing (in the unlikely event the conflict happens with a human or is a one off, vs the more common continual updates of the kubelets/rcs).

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/9569/files#r32160088